mapping.go

```diff
@@ -129,9 +146,29 @@ func (o *mapAssignmentKind) String() string {
 	if o.ElemDirect {
 		s += " (direct)"
 	}
+	if o.ElemConvert {
+		s += " (convert)"
+	}
 	return s
 }
 
+func convertibleButNotIdentical(typ, typeDecode types.Type) bool {
+	// Only allow this for basic and named.
+
+	switch typeDecode.(type) {
+	case *types.Basic, *types.Named:
```

Comment:
Basic: stuff like `string` or `int`
Named: stuff like `Label` in `type Label string`